Assets/Scripts/Player/PlayerHandDetection.cs

```diff
@@ -3,19 +3,72 @@
 using UnityEngine;
 
 public class PlayerHandDetection : MonoBehaviour {
+	[SerializeField] private GameObject interactArrowPrefab;
 	private Interactable detectedInteractItem;
+	private List<Interactable> interactableList = new List<Interactable>();
+	private GameObject interactArrow;
+	private InteractArrow interactArrowScript;
+	private float lowestDistance = 0;
+
 	public Interactable DetectedInteractItem => detectedInteractItem;
 
-	public void OnTriggerStay2D(Collider2D collider) {
-		Interactable interactable = collider.GetComponent<Interactable>();
+	private void Start() {
+		interactArrow = Instantiate(interactArrowPrefab);
+		interactArrowScript = interactArrow.GetComponent<InteractArrow>();
+		interactArrow.SetActive(false);
+	}
+
+	private void OnTriggerEnter2D(Collider2D other) {
+		Interactable interactable = other.GetComponent<Interactable>();
 		if (interactable != null) {
-			detectedInteractItem = interactable;
+			interactableList.Add(interactable);
+			interactArrow.SetActive(true);
 		}
 	}
 
-	public void OnTriggerExit2D(Collider2D collider) {
-		if (collider.GetComponent<Interactable>() != null) {
-			detectedInteractItem = null;
+	private void Update() {
+		if (interactableList.Count == 0)
+			return;
+
+		SetClosestItem();
+	}
+
+	private void OnTriggerExit2D(Collider2D other) {
+		Interactable interactable = other.GetComponent<Interactable>();
+		if (interactable != null) {
+			interactableList.Remove(interactable);	
+			
+			if (interactableList.Count == 0) {
+				detectedInteractItem = null;
+				lowestDistance = 0;
+				interactArrow.SetActive(false);
+			}
+			else {
+				detectedInteractItem = null;
+				SetClosestItem();
+			}
+		}	
+	}
+
+	private void SetClosestItem() {
+		lowestDistance = 0;
+		Interactable closestItem = detectedInteractItem;
+		for (int i = 0; i < interactableList.Count; i++) {
+			Vector3 vectDist = transform.position - interactableList[i].transform.position;
+			float distance = vectDist.sqrMagnitude;
+
+			Debug.DrawLine(interactableList[i].transform.position, transform.position);
+
+			if (distance < lowestDistance || lowestDistance == 0) {
+				lowestDistance = distance;
+				closestItem = interactableList[i];
+			}
 		}
+		if (closestItem == detectedInteractItem) 
+			return;
+
+		interactArrow.transform.position = closestItem.transform.position + new Vector3(0, closestItem.InteractableArrowHeight, 0);
+		interactArrowScript.Initialize();
+		detectedInteractItem = closestItem;
 	}
```

Comment:
Cleaned it up a bit.
```suggestion
	private void SetClosestItem() {
		float lowestDistance = float.PositiveInfinity;
		Interactable closestItem = null;

		foreach (Interactable interactable in interactableList) {
			float distance = (transform.position - interactable.transform.position).sqrMagnitude;

			if (distance < lowestDistance) {
				lowestDistance = distance;
				closestItem = interactable;
			}
		}

		if (closestItem == null)
			return;

		interactArrow.transform.position = closestItem.transform.position + new Vector3(0, closestItem.InteractableArrowHeight, 0);
		interactArrowScript.Initialize();
		detectedInteractItem = closestItem;
	}
```